commons-server/src/main/java/com/navercorp/pinpoint/common/server/bo/serializer/trace/v2/SpanDecoderV0.java

```diff
@@ -180,6 +180,9 @@ public void readSpanValue(Buffer buffer, SpanBo span, SpanDecodingContext decodi
 
         List<SpanEventBo> spanEventBoList = readSpanEvent(buffer, decodingContext, SEQUENCE_SPAN_EVENT_FILTER);
         span.addSpanEventBoList(spanEventBoList);
+        if (bitFiled.isSetAgentName()) {
+            span.setAgentName(buffer.readPrefixedString());
+        }
```

Comment:
@emeroad Here i have two solutions. I don't know which one is prefered.
Solution 1: as the committed. Occupied the last reserved bit of SpanBitFiled to indicate whether the agent name is set or not. Just write or read when the last bit is set. In this case the SpanBitField is used up. It's impossible for future features.

Solution 2: For the read side, just check for buffer.hasRemaining(), if true then readPrefixedString for agent name, the same as https://github.com/pinpoint-apm/pinpoint/pull/7752/files/9df60df694d373415404d69371bc8f3b1c897932..8b843872bb4dd550619544791fd6bad31cf3dc20#diff-355855262013205cf58c6c3dd88f4591dd8d002c8b52461ae6d1ff6dff15f5e8L78-L85 . 
For the write side, just putPrefixedString. 
Is the "remaining" solution just be prefered for compatibility?

Which one would you prefer?

Reply:
In my opinion, there is a way not to store the agentName in the span.

How about including the  agentName in the agentInfo?
The agentName is stored in the agentInfo-table and can be queried using the agentId.

Reply:
Agent names are stored in both tables(agentInfo and trace tables).  I am thinking show agent name in the call tree. I will check if it's possible not to save agent name in span data.